### PR TITLE
Telemetry improvements

### PR DESCRIPTION
**Pull request by vklimontovich**
## What does this PR do?

This PR improves telemetry for cal.com. It moves data collection to [next-collect](https://github.com/jitsucom/next-collect). It removes one 3rd-party call from the client-side rendering

**Environment**: Staging

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Deploy latest version on Vercel and that middleware sends data to telemetry end-point. Set env `TELEMETRY_DEBUG=true` to see (in SSR console) what data points are collected

### CODE REVIEW

**Review by PeerRich — APPROVED** [review]
thank you a ton!

File: apps/web/package.json
Comment (leog) [inline apps/web/package.json:83]: Is this safe to use? it's using `canary` version corresponding to an alpha version 🤔

**Review by leog — CHANGES_REQUESTED** [review]
LGTM, although I added a pending comment to agree upon before considering ready to merge

File: .env.example
Comment (zomars) [inline .env.example:57]: Did the key changed? Or this this unintentional?

File: apps/web/package.json
Comment (zomars) [reply apps/web/package.json:83]: He's the library author @leog so I assume he knows what he's doing 🤔

File: apps/web/pages/_app.tsx
Comment (zomars) [inline apps/web/pages/_app.tsx:44]: Why was this removed?

**Review by zomars — COMMENTED** [review]
Code looks great and tidy. Left some Qs to address before merging

File: apps/web/package.json
Comment (leog) [reply apps/web/package.json:83]: Oh I didn’t realize. To my point, shouldn’t we wait for a stable release to avoid issues @vklimontovich? I had very bad experiences when using unstable releases, but if we are OK with this, never mind me.

Comment (vklimontovich) [reply apps/web/package.json:83]: Yes, it safe. At the moment the latest canary is stable. We haven't figured out release policy yet. We'll do it in next few days, and I'll send an additional PR to switch to latest stream. Until then we'll not going to publish any canaries anyway

File: .env.example
Comment (vklimontovich) [reply .env.example:57]: Yes, it is intentional. Since all events are sent from server-side, there's no need to expose the key to client rendering

File: apps/web/pages/_app.tsx
Comment (vklimontovich) [reply apps/web/pages/_app.tsx:44]: Seems like an artifact of merge conflict resolving, it's not intentional

File: .env.example
Comment (zomars) [reply .env.example:57]: Sorry @vklimontovich I0ve meant the key VALUE itself changed. Was it regenerated?

Comment (vklimontovich) [reply .env.example:57]: No, it's the same API, but different key (used for server-to-server communication with Jitsu)

Comment (zomars) [reply .env.example:57]: Gotcha!

File: apps/web/pages/_app.tsx
Comment (zomars) [reply apps/web/pages/_app.tsx:44]: Let's restore this component and we're good to merge 👍🏽

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Thank you for the contribution @vklimontovich !!

I don't see the calls going from the server after adding TELEMETRY_DEBUG=true flag. Enabled proxy as well for terminal and still doesn't see the calls. I am sure I am missing something. Does it work for others ?

Also, left a few comments

**Review by leog — APPROVED** [review]

File: apps/web/lib/telemetry.ts
Comment (vklimontovich) [reply apps/web/lib/telemetry.ts:8]: Adding this here won't fire event on each API call. See nextCollectBasicSettings => eventTypes => `{ "/api*": null }` - automated tracking of api_call events is disabled. You could delete it, but I'd keep it for future use to keep nomenclature consistent

File: apps/web/pages/team/[slug].tsx
Comment (vklimontovich) [reply apps/web/pages/team/[slug].tsx:39]: It could. I kept it for compatibility. However, I don't think duplicated events is a problem. Do you could sum(page_views) as opposed to distinct page views?

File: apps/web/components/booking/pages/BookingPage.tsx
Comment (vklimontovich) [reply apps/web/components/booking/pages/BookingPage.tsx:337]: You're right, will fix it

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**
